packages/vega-parser/src/parsers/guides/axis-grid.js

```diff
@@ -37,26 +39,100 @@ export default function(spec, config, userEncode, dataRef, band) {
     round:  _('tickRound')
   };
 
-  if (orient === Top || orient === Bottom) {
-    u = 'x';
-    v = 'y';
-    s = 'height';
-  } else {
-    u = 'y';
-    v = 'x';
-    s = 'width';
-  }
-  v2 = v + '2';
+  if (isSignal(orient)) {
+    val = {}
+    for (var key of Object.keys(tickPos)) {
+      val[key] = topOrBottomAxisExpr(orient.signal, tickPos[key],)
+    }
+
+    update['x'] = enter['x'] = {
+      scale: topOrBottomAxisExpr(orient.signal, tickPos.scale, vscale ? vscale : undefined),
+      field: topOrBottomAxisExpr(orient.signal, tickPos.field, undefined),
+      band: topOrBottomAxisExpr(orient.signal, tickPos.band, undefined),
+      extra: topOrBottomAxisExpr(orient.signal, tickPos.extra, undefined),
+      offset: topOrBottomAxisExpr(orient.signal, tickPos.offset, offset),
+      round: topOrBottomAxisExpr(orient.signal, tickPos.round, undefined),
+      range: topOrBottomAxisExpr(orient.signal, undefined, vscale ? 0 : undefined),
+      mult: topOrBottomAxisExpr(orient.signal, undefined, vscale ? sign : undefined),
+      value: topOrBottomAxisExpr(orient.signal, undefined, vscale ? undefined : 0)
+    };
 
-  update[u] = enter[u] = exit[u] = tickPos;
+    update['x2'] = enter['x2'] = {
+      scale: topOrBottomAxisExpr(orient.signal, tickPos.scale, vscale ? vscale : undefined),
```

Comment:
You can probably write a function to create these blocks.

Reply:
Good suggestion, done! Thanks